src/js/node/http.ts

```diff
@@ -953,11 +957,43 @@ class OutgoingMessage extends Writable {
   }
 }
 
+function emitCloseNT(self) {
+  if (!self._closed) {
+    self.destroyed = true;
+    self._closed = true;
+    self.emit("close");
+  }
+}
+
+function onServerResponseClose() {
+  // EventEmitter.emit makes a copy of the 'close' listeners array before
+  // calling the listeners. detachSocket() unregisters onServerResponseClose
+  // but if detachSocket() is called, directly or indirectly, by a 'close'
+  // listener, onServerResponseClose is still in that copy of the listeners
+  // array. That is, in the example below, b still gets called even though
+  // it's been removed by a:
+  //
+  //   const EventEmitter = require('events');
+  //   const obj = new EventEmitter();
+  //   obj.on('event', a);
+  //   obj.on('event', b);
+  //   function a() { obj.removeListener('event', b) }
+  //   function b() { throw "BAM!" }
+  //   obj.emit('event');  // throws
+  //
+  // Ergo, we need to deal with stale 'close' events and handle the case
+  // where the ServerResponse object has already been deconstructed.
+  // Fortunately, that requires only a single if check. :-)
+  if (this._httpMessage) {
+    emitCloseNT(this._httpMessage);
+  }
+}
+
 let OriginalWriteHeadFn, OriginalImplicitHeadFn;
-class ServerResponse extends Writable {
+class ServerResponse extends OutgoingMessage {
   declare _writableState: any;
 
-  constructor({ req, reply }) {
+  constructor(req, reply) {
```

Comment:
this is what im referring to. the other changes i think are ok

Reply:
But this is the main problem that this PR solves
many people face this problem, also many examples are related
the problem is that now ServerResponse is not compatible with node
https://github.com/nodejs/node/blob/v21.3.0/lib/_http_server.js#L195

Reply:
I think this change is necessary
if such a change is a concern, I can make this parameters more compatible with the current version of bun

Reply:
ah i see. this is okay then.